components/support/jwcrypto/src/error.rs

```diff
@@ -22,4 +22,25 @@ pub enum JwCryptoError {
     JsonError(#[from] serde_json::Error),
     #[error("UTF8 decode error: {0}")]
     UTF8DecodeError(#[from] std::string::FromUtf8Error),
+    #[error("InvalidKey")]
+    InvalidKey,
+    #[error("EmptyCyphertext")]
+    EmptyCyphertext,
+}
+
+/// Error for the EncryptorDecryptor struct
+///
+/// The Display implementation will include a description of what was being encrypted/decrypted
+/// when the error happened
+///
+/// This has a nice interaction with `error_support::ErrorHandling`. Crates will typically define
+/// an internal error with a variant that wraps `EncryptorDecryptorError`.  As long as the Display
+/// implementation for that variant includes the `EncryptorDecryptorError` string, then the
+/// description will be included in error reports to Sentry and other systems.
+#[derive(Error, Debug)]
+#[error("{from} ({description})")]
+pub struct EncryptorDecryptorError {
+    // Underlying error
+    pub from: JwCryptoError,
+    pub description: String,
```

Comment:
Wow! This is neat!

How do we feel this works vs having `EncryptorDecryporError` be "just" another variant in `JwCryptoError`?

That way we can even remove the `E` type parameter from `EncryptorDecryptor` and have each function in `EncryptorDecryptor` just return `Result<T, JwCryptoError>` and it's the consumer's job to turn that into its own error (and report it)

i.e: I realize this sounds a lot like the opposite of what I was suggesting earlier so sorry for that, curious what you think works best here

Reply:
I like the idea and I don't even think we need a new variant.  But many of the current variants would need to store `description` strings.

The reason for a different error type is that I was trying to leave the existing API alone and define a new one on-top of it.  `fxa-client` is still using the current API and I don't think I want to change that code yet.

How about as a follow up as part of the fxa work?